### PR TITLE
fix(grammar): allow bare bussin; void return statements (#283)

### LINKED ISSUE #283: [Bug]: bare bussin; (void return) is a parse error — skibidi functions can't return early
### Description

A bare `bussin;` (return with no value) is a **parse error**, so a `skibidi` (void) function has no way to return early. This contradicts the interpreter, which was clearly built to support a value-less return: `interpreter_visit_return_statement` explicitly handles the `node->data.op.left == NULL` case and calls `handle_return_statement(NULL)` (`interpreter.c:824-835`).

The gap is purely in the grammar. `return_statement` has a single production that *requires* an expression:

```
return_statement:
    BUSSIN expression
        { $$ = create_return_node($2); }
    ;
```
(`lang.y:1907`)

There is no `BUSSIN SEMICOLON` alternative, so `bussin;` can never be produced.

### Minimal reproduction code

```c
skibidi noop() {
    bussin;              🚽 early return from a void function
}
skibidi main {
    noop();
    yapping("ok");
}
```

### Expected behavior

Parses and runs, printing `ok`. A `skibidi` (void) function can `bussin;` to return early, mirroring C's `return;`.

### Actual behavior

```shell
Error: syntax error, unexpected SEMICOLON at line 1
Parsing failed
```

### Brainrot version / commit

dcddf726a3f818c9d4e1eadc040e7a91aa9ba99f

### Operating system

WSL2 (Ubuntu), x86-64

### Additional context

Fix is a one-line grammar addition — a `BUSSIN SEMICOLON` alternative in `return_statement` that builds a return node with a NULL expression (`create_return_node(NULL)`), which the interpreter already handles. Distinct from #246, which is about `bussin <expr>` in `main` not halting execution.

### PR DESCRIPTION
**Pull request by Aditya-233**
## Description

Fixes an issue where a bare `bussin;` (void return with no expression) failed to parse because the Bison grammar only accepted `BUSSIN expression` under `return_statement`.

The interpreter was already implemented to support value-less return (`interpreter_visit_return_statement` in `interpreter.c` explicitly handles `node->data.op.left == NULL` by calling `handle_return_statement(NULL)`).

This change adds the `BUSSIN` alternative to `return_statement` in `lang.y` to build a return node with a `NULL` expression (`create_return_node(NULL)`).

### Tests and Stress Testing
Added regression test fixture `test_cases/bare_bussin_void_return.brainrot` (and matching expected output in `tests/expected_results.json`) covering:
- Bare `bussin;` in a noop void function
- Conditional early return from a void function (`edgy (...) { bussin; }`)
- Recursive void function using `bussin;` for base case termination
- Scope and local variable cleanup when `bussin;` exits early
- Bare `bussin;` at the end of `skibidi main`

Additionally, stress tested across:
- Nested conditional branches (`edgy` / `sus`)
- Deeply nested scopes and structs
- Multiple sequential void calls with early exits
- Full valgrind memory leak suite (0 leaks, 0 errors)
- Full compiler test harness (477 passed, 5 skipped for optional raylib)

## Related Issue

Fixes #283

## Type of Change

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Documentation update
- [ ] Performance improvement
- [ ] Refactor

## Checklist

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have documented my changes in the code or documentation
- [x] I have added tests that prove my change works, at the lowest appropriate layer (`test_cases/*.brainrot` for language-visible behavior; a host-side C test wired into `make test` for internal APIs Brainrot source cannot reach). Required for every bug fix, feature, refactor, performance change, and breaking change — not optional, not "if applicable". See CONTRIBUTING.md ("TESTS ARE MANDATORY").
- [x] Those tests cover the happy path, the original bug (for fixes), error cases, edge cases, **and** adversarial cases. If the existing harness could not reach the behavior, I extended it or added a lower-level test.
- [ ] If this PR cannot affect program behavior (docs/comments/license only), I explained that in the Description instead of skipping the items above in silence
- [x] I have run `make format-check` locally (or `make format` to fix)
- [x] I have run the unit tests locally
- [x] I have run the valgrind memory tests locally
- [x] All new and existing tests pass

### CODE REVIEW

File: lang.y [hunk 1]
```diff
@@ -1991,8 +1991,10 @@ error_statement:
     ;
 
 return_statement:
-    BUSSIN expression
+      BUSSIN expression
         { $$ = create_return_node($2); }
+    | BUSSIN
```
Comment (leo-aa88) [inline lang.y:1996]: Correct and safe. `create_return_node(NULL)` sets `data.op.left = NULL`, which `interpreter_visit_return_statement` and `handle_return_statement`'s `!expr` arm already handle, and the semantic analyzer's return visitor is NULL so nothing dereferences the missing expression. Verified: `bison -Wcounterexamples` reports zero conflicts before and after (SEMICOLON is not in FIRST(expression), so the optional-expression form is unambiguous), and the checked-in fixture that passes here fails to parse on main's old grammar (`unexpected SEMICOLON`). Non-blocking: this also makes bare `bussin;` legal in a value-returning function, where the `!expr` arm returns a stale `current_return_value` payload, but that is the same pre-existing gap as falling off the end of such a function, not something this PR introduces.

**Review by leo-aa88 — APPROVED** [review]
LGTM.

I built this and verified it end to end rather than trusting the description.

**The fix is correct and minimal.** `return_statement` only accepted `BUSSIN expression`, so a bare `bussin;` was a hard parse error even though the interpreter already handled a value-less return (`interpreter_visit_return_statement` branches on `node->data.op.left == NULL`, and `handle_return_statement` has an `!expr` arm). Adding `| BUSSIN { create_return_node(NULL); }` activates that existing path. `create_return_node(NULL)` sets `data.op.left = NULL` safely, and the semantic analyzer registers a NULL return visitor, so the value-less node is never dereferenced during analysis.

**No new grammar conflicts.** I regenerated the parser with `bison -Wcounterexamples` on both main and this branch: zero conflicts before, zero after. The optional-expression alternative is unambiguous because SEMICOLON (the FOLLOW of `return_statement`) is not in the FIRST set of `expression`.

**The fixture is a real regression test.** On this branch the checked-in `test_cases/bare_bussin_void_return.brainrot` prints exactly the expected `noop passed / positive: 10 / count: 3 / count: 2 / count: 1 / liftoff / locals exit passed`. I then rebuilt the interpreter with main's old grammar and the same fixture fails to parse (`syntax error, unexpected SEMICOLON at line 19`), so it genuinely pins the bug. It covers a noop void function, conditional early return, a recursive base case, early return with live locals/struct/array in scope, and bare `bussin;` at the end of `skibidi main`. Expected-output entry matches byte for byte, and the ASan/UBSan build runs it leak-clean.

Full `pytest` run is green (483 passed) with no regression.

**One non-blocking observation, not introduced by this PR.** Allowing bare `bussin;` everywhere also makes it legal inside a value-returning function (e.g. `rizz f() { bussin; }`), where `handle_return_statement`'s `!expr` arm sets `has_value = true` and the declared type but never writes `value.*`, so the caller reads whatever was last left in the global `current_return_value` slot. I confirmed `rizz bad(...) { rizz t = side(999); bussin; }` returns a stale `999`. This is NOT a regression: falling off the end of a value-returning function with no `bussin` at all already returns the same stale value today, so this is the pre-existing "no missing-return diagnostic" gap, reached by one more syntactic path. Worth a follow-up if the project ever adds return-value checking; not a reason to hold this fix.

Ship it.
